src/event-emitter.js

```diff
@@ -26,11 +26,18 @@ define(['lodash-amd/modern/arrays/pull',
   };
 
   EventEmitter.prototype.trigger = function (eventName, args) {
-    var listeners = this._listeners[eventName] || Immutable.Set();
 
-    listeners.forEach(function (listener) {
-      listener.apply(null, args);
-    });
+    //fire events like my:custom:event -> my:custom -> my
+    var events = eventName.split(':');
+    while(events.length){
```

Comment:
Please avoid using integers as booleans for clarity, [as per styleguide](https://github.com/guardian/flexible-content/wiki/JavaScript-Style-Guide#syntax).